common/OpTestSSH.py

```diff
@@ -155,52 +159,133 @@ def get_console(self):
 
         return self.console
 
-    def run_command(self, command, timeout=60):
-        console = self.get_console()
-        if self.prompt:
-            prompt = self.prompt
+    def set_env(self, console):
+        set_env_list = []
+        if self.use_default_bash:
+          console.sendline("exec bash --norc --noprofile")
+        expect_prompt = self.build_prompt() + "$"
+        console.sendline('PS1=' + self.build_prompt())
+        console.expect(expect_prompt)
+        combo_io = (console.before + console.after).lstrip()
+        set_env_list += combo_io.splitlines()
+        # remove the expect prompt since matched generic #
+        del set_env_list[-1]
+        return set_env_list
+
+    def try_sendcontrol(self, console, command):
+        console.sendcontrol('c')
+        try_list = []
+        try_list += self.set_env(console)
+        expect_prompt = self.build_prompt() + "$"
+        # we need the env to get the expect_prompt
+        rc = console.expect([expect_prompt, pexpect.TIMEOUT, pexpect.EOF], 10)
```

Comment:
After setting the prompt(in set_env function) we are expecting expect_prompt for two times(once in set_env and another in try_sendcontrol), that never going to happen and hence always the timeout occurs for set prompt functionality.


I think for timeout case:
    a. run the command
    b. wait for timeout
    c. once we hit pexpect.TIMEOUT 
            call try_sendcontrol()--> set env back , if any timeout occurs here, this is a ENV timeout, not the actual command timeout.
     d. Once try_sendcontrol got success raise the actual run_command TIMEOUT exception.

Reply:
Fixed

Reply:
jfyi - I currently do not have machine to test out fix in this error path.